### PR TITLE
FOUR-16001 Implement resources and apiClient to use v1.1 in charts

### PR DESCRIPTION
**Pull request by caleeli**
## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-16001

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:next

### CODE REVIEW

File: ProcessMaker/Http/Resources/V1_1/TaskResource.php [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+    protected static $defaultFieldsFor = [
+        'user' => ['id', 'firstname', 'lastname', 'email', 'username', 'avatar'],
+        'requestor' => ['id', 'first_name', 'last_name', 'email'],
+        'processRequest' => ['id', 'process_id', 'status'],
+        'draft' => ['id', 'task_id', 'data'],
+        // 'component' => ['id', 'name', 'type', 'version'],
+        'screen' => ['id', 'config'],
+        // 'requestData',
+        // 'loopContext',
```
Comment (rodriquelca) [inline ProcessMaker/Http/Resources/V1_1/TaskResource.php:54]: remove code comments

File: ProcessMaker/Http/Resources/V1_1/TaskResource.php [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    {
+        if (!method_exists($model, $relationship)) {
+            return false;
+        }
+        $relationshipObject = $model->$relationship();
+        if (!($relationshipObject instanceof Relation)) {
+            return false;
+        }
+
+        // $singleRelationship = $relationshipObject instanceof HasOne || $relationshipObject instanceof BelongsTo;
```
Comment (rodriquelca) [inline ProcessMaker/Http/Resources/V1_1/TaskResource.php]: remove this line

File: ProcessMaker/Http/Resources/V1_1/TaskResource.php [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        'element_id',
+        'due_at',
+    ];
+
+    protected static $defaultFieldsFor = [
+        'user' => ['id', 'firstname', 'lastname', 'email', 'username', 'avatar'],
+        'requestor' => ['id', 'first_name', 'last_name', 'email'],
+        'processRequest' => ['id', 'process_id', 'status'],
+        'draft' => ['id', 'task_id', 'data'],
+        // 'component' => ['id', 'name', 'type', 'version'],
```
Comment (rodriquelca) [inline ProcessMaker/Http/Resources/V1_1/TaskResource.php:51]: remove commented code

File: ProcessMaker/Http/Resources/V1_1/TaskResource.php [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        'screen' => ['id', 'config'],
+        // 'requestData',
+        // 'loopContext',
+        // 'definition'
+        // 'bpmnTagName',
```
Comment (rodriquelca) [inline ProcessMaker/Http/Resources/V1_1/TaskResource.php:56]: clean this section

File: ProcessMaker/Http/Resources/V1_1/TaskResource.php (hunk 2 above)
Comment (caleeli) [reply ProcessMaker/Http/Resources/V1_1/TaskResource.php]: fixed

File: ProcessMaker/Http/Resources/V1_1/TaskResource.php [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    {
+        foreach (self::$defaultFields as $field) {
+            $query->addSelect($field);
+        }
+
+        $include = $request->query('include', []);
+        if ($include) {
+            $include = explode(',', $include);
+        }
+        $include[] = 'process';
```
Comment (rodriquelca) [inline ProcessMaker/Http/Resources/V1_1/TaskResource.php:119]: hardcoded maybe we can use constants

File: ProcessMaker/Http/Resources/V1_1/TaskResource.php [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+        if ($include) {
+            $include = explode(',', $include);
+        }
+
+        foreach (self::$includeMethods as $key) {
+            if (!in_array($key, $include)) {
+                continue;
+            }
+
+            $method = "include" . ucfirst($key);
```
Comment (rodriquelca) [inline ProcessMaker/Http/Resources/V1_1/TaskResource.php:158]: hardcoded include string

File: ProcessMaker/Http/Resources/V1_1/TaskResource.php [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        $model = $query->first();
+        foreach (self::$includeMethods as $key) {
+            if (!in_array($key, $include)) {
+                continue;
+            }
+
+            self::addRelationship($model, $key);
+        }
+
+        return new static($query->first());
```
Comment (rodriquelca) [inline ProcessMaker/Http/Resources/V1_1/TaskResource.php:143]: maybe we can improve $query->first() was called two times


```

    $model = $query->first();
        foreach (self::$includeMethods as $key) {
            if (in_array($key, $include)) {
                self::addRelationship($model, $key);
            }
        }

        return new static($model);
```

Comment (caleeli) [reply ProcessMaker/Http/Resources/V1_1/TaskResource.php:143]: Very good catch!

File: ProcessMaker/Http/Resources/V1_1/TaskResource.php (hunk 5 above)
Comment (caleeli) [reply ProcessMaker/Http/Resources/V1_1/TaskResource.php]: Added a const for the default includes

File: ProcessMaker/Http/Resources/V1_1/TaskResource.php (hunk 6 above)
Comment (caleeli) [reply ProcessMaker/Http/Resources/V1_1/TaskResource.php]: Fixed

File: resources/js/bootstrap.js [hunk 8]
```diff
@@ -233,6 +233,17 @@ if (token) {
 }
 
 window.ProcessMaker.apiClient.defaults.baseURL = "/api/1.0/";
+// If requested url starts with "/api/1.1/" use "/api/1.1/"
+window.ProcessMaker.apiClient.interceptors.request.use((config) => {
+  if (config.url.startsWith("/api/1.1/")) {
```
Comment (rodriquelca) [inline resources/js/bootstrap.js:238]: hard coded  "/api/1.1/"

**Review by rodriquelca — APPROVED** [review]

File: resources/js/bootstrap.js (hunk 8 above)
Comment (caleeli) [reply resources/js/bootstrap.js]: Fixed

File: resources/js/bootstrap.js [hunk 9]
```diff
@@ -232,7 +232,30 @@ if (token) {
   console.error("CSRF token not found: https://laravel.com/docs/csrf#csrf-x-csrf-token");
 }
 
+// Setup api versions
+const apiVersionConfig = [
+  { version: "1.0", baseURL: "/api/1.0/" },
+  { version: "1.1", baseURL: "/api/1.1/" },
+];
+
 window.ProcessMaker.apiClient.defaults.baseURL = "/api/1.0/";
```
Comment (julceslauhub) [inline resources/js/bootstrap.js:241]: use here the new constant

**Review by rodriquelca — APPROVED** [review]

File: resources/js/bootstrap.js (hunk 9 above)
Comment (caleeli) [reply resources/js/bootstrap.js:241]: Fixed

**Pull request merged**